src/script/miniscript.cpp

```diff
@@ -11,6 +11,16 @@
 #include <assert.h>
 
 namespace miniscript {
+
+bool IsTapscript(MiniscriptContext ms_ctx)
+{
+    switch (ms_ctx) {
+        case MiniscriptContext::P2WSH: return false;
+        case MiniscriptContext::TAPSCRIPT: return true;
+    }
+    assert(false);
```

Comment:
iirc, `assert` crashes the node. wouldn't it be better to use the `Assume` macro?

Reply:
This specific case is the very common match-all-variants-and-assert pattern so i think it's good like that. It's also documented this way in our style guidelines: https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#c-data-structures.